var/spack/repos/builtin/packages/py-pip/package.py

```diff
@@ -104,7 +106,32 @@ def url_for_version(self, version):
             python_tag = "py2.py3"
         return url.format(python_tag, version)
 
+    def setup_build_environment(self, env):
+        """Specify where the user site-packages directory should be"""
+        if sys.platform == "win32":
+            env.set("PYTHONUSERBASE", self.spec.prefix)
+
     def install(self, spec, prefix):
+        # Pip must be bootstrapped on windows using ensurepip. When setting
+        # up the build environment, the PYTHONUSERBASE variable is injected
+        # into the environment, set to the spec prefix. Then, in conjunction
+        # with the --user flag, pip is installed into the appropriate directory.
+        # Once bootstrapped, we can install the desired version of pip from
+        # the wheel.
+        if sys.platform == "win32":
+            python("-m", "ensurepip", "--user")
+            whl = self.stage.archive_file
+            upgrade_cmd = ["-m", "pip", "install", "--user", whl]
+            python(*upgrade_cmd)
+            # Need to fix the directory name to "Lib" in order for the python package to correctly
+            # set the PYTHONPATH variable so the pip is usable because this is the structure that
+            # other python packages are installed with.
+            # TODO (smillie) Does this need to be protected in a try/except?
+            orig_pydir = glob.glob(os.path.join(self.spec.prefix, "Python3*"))[0]
+            new_pydir = os.path.join(self.spec.prefix, "Lib")
```

Comment:
I think it would make sense for the destination to be `self.python_platlib` (which makes it more explicit since the package.py for Python is setting PYTHONPATH based on `platlib`).

I think it would also make sense to calculate the source relative to an expected file in the installation, vs. a preconceived base.

Reply:
This is no longer relevant when using pip.pyz